controllers/devworkspace/controller_test.go

```diff
@@ -382,18 +357,20 @@ func TestCustomResourceFinalization(t *testing.T) {
 	}
 
 	// try to delete the manager and check that the configmap disallows that and that the status of the manager is updated
-	manager.DeletionTimestamp = &metav1.Time{Time: time.Now()}
-	err = cl.Update(ctx, &manager)
-	if err != nil {
-		t.Fatalf("Failed to update the manager in the fake client: %s", err)
-	}
+	// It is not possible to update readonly field
+	//manager.DeletionTimestamp = &metav1.Time{Time: time.Now()}
```

Comment:
Would it be possible to simulate deletion by calling `client.Delete()` instead of manually setting `.DeletionTimestamp`?

Does the fake client automatically set `.DeletionTimestamp` when `Delete()` is called, or would we need to mock that behavior manually for finalizer testing?